staging/src/k8s.io/api/admission/v1alpha1/types.go

```diff
@@ -20,62 +20,73 @@ import (
 	authenticationv1 "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// AdmissionReview describes an admission request.
+// AdmissionReview describes an admission review request/response.
 type AdmissionReview struct {
 	metav1.TypeMeta `json:",inline"`
-	// Spec describes the attributes for the admission request.
-	// Since this admission controller is non-mutating the webhook should avoid setting this in its response to avoid the
-	// cost of deserializing it.
+	// Request describes the attributes for the admission request.
+	Request AdmissionRequest `json:"request,omitempty" protobuf:"bytes,1,opt,name=request"`
```

Comment:
Should `Request` be optional? 

Do you make it `required` because the UID/Kind/Resource are required even if the `AdmissionReview` is a response from the webhook?

Reply:
I made it non optional so as you suggest fields like UID/Kind/Resource are present on the response and could be validated by the KAS.

Reply:
Things required in the response should probably be in the response portion of the object. The unique identifier makes particular sense.